admin-ui/plugins/auth-server/components/Agama/AgamaAliasListPage.js

```diff
@@ -107,16 +135,40 @@ function AliasesListPage() {
     },
   })
 
-  const handleSubmit = useCallback(
-    (values) => {
+  const handleSubmit = useCallback(() => {
+    setShowCommitDialog(true)
+  }, [])
+
+  const handleDeleteAccept = (userMessage) => {
+    if (!rowToDelete) return
+    const userAction = {}
+    const postBody = {}
+    const value = { ...configuration?.acrMappings }
+    delete value[rowToDelete.mapping]
+    postBody['requestBody'] = [
+      {
+        path: '/acrMappings',
+        value: value,
+        op: configuration?.acrMappings ? 'replace' : 'add',
+      },
+    ]
+    buildPayload(userAction, userMessage, postBody)
+    dispatch(patchJsonConfig({ action: userAction }))
+    if (rowToDelete.resolve) rowToDelete.resolve(true)
+    setShowDeleteDialog(false)
+    setRowToDelete(null)
+  }
+
+  const handleCommitAccept = useCallback(
+    (userMessage) => {
       const userAction = {}
       const postBody = {}
       let value = configuration.acrMappings
 
       if (isEdit) {
         delete value[selectedRow.mapping]
       }
-      value = { ...value, [values.mapping]: values.source }
+      value = { ...value, [formik.values.mapping]: formik.values.source }
```

Comment:
Mutating configuration object directly

The code directly mutates the `value` object which references `configuration.acrMappings`. This could cause unintended side effects and React state update issues.


Create a new object instead of mutating the existing one:

```diff
-      let value = configuration.acrMappings
+      let value = { ...configuration.acrMappings }

       if (isEdit) {
         delete value[selectedRow.mapping]
       }
       value = { ...value, [formik.values.mapping]: formik.values.source }
```